src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -191,64 +188,86 @@
-          report (some-> report update-report-run-fields)
-          factset (some-> factset (update-factset rand-percentage))]
+    (let [catalog (some->> catalog (maybe-tweak-catalog rand-percentage))
+          factset (some->> factset (maybe-tweak-factset rand-percentage))]
       ;; Submit the catalog and reports in separate threads, so as to not
       ;; disturb the world-loop and otherwise distort the space-time continuum.
       (when catalog
         (future
           (try
-            (client/submit-catalog base-url 6 (json/generate-string catalog))
+            (->> catalog
+                 update-producer-timestamp
+                 json/generate-string
+                 (client/submit-catalog base-url 6))
             (log/infof "[%s] submitted catalog" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit catalog: %s" host e)))))
       (when report
         (future
           (try
-            (client/submit-report base-url 5 (json/generate-string report))
+            (some->> report
+                     update-report
+                     json/generate-string
+                     (client/submit-report base-url 5))
             (log/infof "[%s] submitted report" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit report: %s" host e)))))
       (when factset
         (future
           (try
-            (client/submit-facts base-url 4 (json/generate-string factset))
+            (->> factset
+                 update-producer-timestamp
+                 json/generate-string
+                 (client/submit-facts base-url 4)) 
             (log/infof "[%s] submitted factset" host)
             (catch Exception e
               (log/errorf "[%s] failed to submit factset: %s" host e)))))
       (assoc state
-        :lastrun clock
-        :catalog catalog))))
+             :lastrun clock
+             :factset factset
+             :catalog catalog))))
 
 (defn update-host
   "Submit a `catalog` for `hosts` (when present), possibly mutating it before
    submission.  Also submit a report for the host (if present). This is
    similar to timed-update-host, but always sends the update (doesn't run/skip
    based on the clock)"
-  [{:keys [host lastrun catalog report factset base-url run-interval rand-percentage] :as state}]
-  (let [catalog (some-> catalog update-catalog (maybe-tweak-catalog rand-percentage))
-        report (some-> report update-report-run-fields)
-        factset (some-> factset (update-factset rand-percentage))]
-    (when catalog (client/submit-catalog base-url 6 (json/generate-string catalog)))
-    (when report (client/submit-report base-url 5 (json/generate-string report)))
-    (when factset (client/submit-facts base-url 4 (json/generate-string factset)))
-    (assoc state :catalog catalog)))
+  [{:keys [catalog report factset] :as state} base-url rand-percentage]
```

Comment:
correct, I took them out of state and passed them down from the `cli` since they don't change.